### ISSUE TITLE
How to get rid of duplicate metrics ？

### ISSUE THREAD

**Issue by yzhengwei** [VictoriaMetrics/VictoriaMetrics#3103]
I used VictoriaMetrics V1.69.0 to collect the Kubernetes indicator. When I looked up the metrics, I found two very similar metrics, except that  `label_topology_ebs_csi_aws_com_zone` label was not the same. Why? How do you get rid of one of them when you collect metrics?

```
kube_node_labels{account="shareitwww", cluster="BDP-eks-bigdata-use1-prod", instance="kube-state-metrics.kube-admin:8080", job="kube-state-metrics", label_beta_kubernetes_io_arch="amd64", label_beta_kubernetes_io_instance_type="m5.2xlarge", label_beta_kubernetes_io_os="linux", label_billing_entry="data_engineering", label_billing_group="bigdata", label_billing_service="spark", label_env="prod", label_failure_domain_beta_kubernetes_io_region="us-east-1", label_failure_domain_beta_kubernetes_io_zone="us-east-1a", label_k8s_io_cloud_provider_aws="5fa15b947deba7ec57b2081119fde648", label_kubernetes_io_arch="amd64", label_kubernetes_io_hostname="ip-172-36-147-48.ec2.internal", label_kubernetes_io_os="linux", label_lifecycle="Ec2Spot", label_node_kubernetes_io_instance_type="m5.2xlarge", label_topology_ebs_csi_aws_com_zone="us-east-1a", label_topology_kubernetes_io_region="us-east-1", label_topology_kubernetes_io_zone="us-east-1a", label_workload="general", node="ip-172-36-147-48.ec2.internal", region="us-east-1"}
kube_node_labels{account="shareitwww", cluster="BDP-eks-bigdata-use1-prod", instance="kube-state-metrics.kube-admin:8080", job="kube-state-metrics", label_beta_kubernetes_io_arch="amd64", label_beta_kubernetes_io_instance_type="m5.2xlarge", label_beta_kubernetes_io_os="linux", label_billing_entry="data_engineering", label_billing_group="bigdata", label_billing_service="spark", label_env="prod", label_failure_domain_beta_kubernetes_io_region="us-east-1", label_failure_domain_beta_kubernetes_io_zone="us-east-1a", label_k8s_io_cloud_provider_aws="5fa15b947deba7ec57b2081119fde648", label_kubernetes_io_arch="amd64", label_kubernetes_io_hostname="ip-172-36-147-48.ec2.internal", label_kubernetes_io_os="linux", label_lifecycle="Ec2Spot", label_node_kubernetes_io_instance_type="m5.2xlarge", label_topology_kubernetes_io_region="us-east-1", label_topology_kubernetes_io_zone="us-east-1a", label_workload="general", node="ip-172-36-147-48.ec2.internal", region="us-east-1"}
```

**Comment by laixintao** [comment]
According to https://docs.victoriametrics.com/keyConcepts.html#what-is-a-metric, the two that you listed are two different metrics, so there is no way to "get rid of them" during collect time, as for vmagent, they are different metrics. (I might be wrong).

The only way I can think of is to use recording rules: https://docs.victoriametrics.com/vmalert.html#rules

**Comment by hagen1778** [comment]
Hi @yzhengwei! 
As @laixintao said, these are two distinct time series for VictoriaMetrics. It just stores what you put in it.
If you think you don't want some of these time series - you can drop them via [relabeling](https://docs.victoriametrics.com/relabeling.html). Using relabeling you can drop certain time series on scrape or on ingestion.

**Comment by yzhengwei** [comment]
> According to https://docs.victoriametrics.com/keyConcepts.html#what-is-a-metric, the two that you listed are two different metrics, so there is no way to "get rid of them" during collect time, as for vmagent, they are different metrics. (I might be wrong).
> 
> The only way I can think of is to use recording rules: https://docs.victoriametrics.com/vmalert.html#rules

These two metrics have different labels, which I know means they are two different metrics. But these two metrics are kind of similar, they're actually the same thing, and there's not just one index like this, there's a lot of them, I can't get rid of them by relabel. Why does Vmagent automatically find such similar metrics ?

**Comment by hagen1778** [comment]
> Why does Vmagent automatically find such similar metrics ?

vmagent is a metrics collector. It is not up to vmagent to decide which metrics to scrape or not. It is up to the user, always.

**Comment by laixintao** [comment]
> But these two metrics are kind of similar, they're actually the same thing

Actually they are not similar at all. `up{foo="bar", hello="world"}` is actually a syntax sugar, it's actually: `{__name__="up", foo="bar", hello="world"}`.

> Why does Vmagent automatically find such similar metrics?

If you think about it, you will find it's very hard to do so(and vmagent shouldn't). For e.g.

```
count{foo="bar1"}
count{foo="bar2"}
```

If you want to merge those two metrics, then how to merge?

- add them up? then what if there is `counter reset to zero` case which handled by `rate/increase/irate`?
- what's the new metrics labels?
- bucket seems similar as well, should I merge them?

So it's really should up to the user do define relabeling.

I am also facing the same issue in my case which I need to see the CPU usage of millions of containers, I am currently working on a massive recording rule evaluation.

FYI, I have tried m3db which can aggregate at scraping time: https://m3db.io/ it might suit what you want. But I tried it, its performance is much worse than aggregating by vmalert + vmselect + vmstorage.
